var/spack/repos/builtin/packages/herwig3/package.py

```diff
@@ -54,11 +62,16 @@ def configure_args(self):
             "--with-madgraph=" + self.spec["madgraph5amc"].prefix,
             "--with-openloops=" + self.spec["openloops"].prefix,
             "--with-gosam-contrib=" + self.spec["gosam-contrib"].prefix,
-            "--with-njet=" + self.spec["njet"].prefix,
-            "--with-vbfnlo=" + self.spec["vbfnlo"].prefix,
             "--with-evtgen=" + self.spec["evtgen"].prefix,
+            "--with-gosam=" + self.spec["py-gosam"].prefix,
         ]
 
+        if self.spec.satisfies("+njet"):
+            args.append("--with-njet=" + self.spec["njet"].prefix)
+
+        if self.spec.satisfies("+vbfnlo"):
+            args.append("--with-vbfnlo=" + self.spec["vbfnlo"].prefix)
```

Comment:
FYI.  You could take advantage of the with-or-without helper on these (see https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#activation-value).

Reply:
I had problems in the past with (hep) packages not handling `--without-X` correctly, so I'd prefer not to use it